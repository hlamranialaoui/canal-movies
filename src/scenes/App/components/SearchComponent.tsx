import React from "react";

interface ISearchProps {
    filter: any;
}

export default function SearchComponent({filter}: ISearchProps) {
    return (
        <div className="bg-img">
            <div className="layer">
                <div className="row w-100">
                    <div className="col-md-2 col-1"/>
                    <div className="col-md-8 col-8">
                        <div className="input-group mb-3">
                            <input type="text"
                                   className="form-control" placeholder="search movie"
                                   onChange={(event) => filter(event.target.value)}
                            />
                            <span className="input-group-text" id="basic-addon2">
                                Search by movie name
                            </span>
                        </div>
                    </div>
                    <div className="col-md-2 col-1"/>
                </div>
            </div>
        </div>
    );
}