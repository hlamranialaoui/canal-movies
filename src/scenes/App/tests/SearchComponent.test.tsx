import React from 'react';
import { cleanup, render } from '@testing-library/react';
import SearchComponent from "../components/SearchComponent";

afterEach(cleanup);

describe('renders canal search component', () => {

    test('renders search component', async () => {
        const { asFragment } = render(
            <SearchComponent
                filter={'originalName'}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
