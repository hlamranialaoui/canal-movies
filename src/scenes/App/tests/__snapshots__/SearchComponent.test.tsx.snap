// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders canal search component renders search component 1`] = `
<DocumentFragment>
  <div
    class="bg-img"
  >
    <div
      class="layer"
    >
      <div
        class="row w-100"
      >
        <div
          class="col-md-2 col-1"
        />
        <div
          class="col-md-8 col-8"
        >
          <div
            class="input-group mb-3"
          >
            <input
              class="form-control"
              placeholder="search movie"
              type="text"
            />
            <span
              class="input-group-text"
              id="basic-addon2"
            >
              Search by movie name
            </span>
          </div>
        </div>
        <div
          class="col-md-2 col-1"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
